starknet_py/net/schemas/rpc.py

```diff
@@ -87,13 +89,18 @@ def make_dataclass(self, data, **kwargs) -> L2toL1Message:
 
 class TransactionReceiptSchema(Schema):
     transaction_hash = Felt(data_key="transaction_hash", required=True)
+    # replaced by execution and finality status in RPC v0.4.0-rc1
     status = StatusField(data_key="status", load_default=None)
+    execution_status = ExecutionStatusField(
+        data_key="execution_status", load_default=None
+    )
+    finality_status = FinalityStatusField(data_key="finality_status", load_default=None)
     block_number = fields.Integer(data_key="block_number", load_default=None)
     block_hash = Felt(data_key="block_hash", load_default=None)
     actual_fee = Felt(data_key="actual_fee", required=True)
     type = TransactionTypeField(data_key="type", load_default=None)
     contract_address = Felt(data_key="contract_address", load_default=None)
-    rejection_reason = fields.String(data_key="status_data", load_default=None)
+    rejection_reason = fields.String(data_key="revert_reason", load_default=None)
```

Comment:
I think we shouldn't rename the field to `rejection_reason`. `rejection_reason` is for rejected transactions and this just introduces more confusion. We could add `revert_reason` to `TransactionReceipt` or have it named here `revert_error` - wdyt?